homeassistant/components/dynalite/__init__.py

```diff
@@ -120,20 +149,27 @@ def num_string(value):
     """Set up a bridge from a config entry."""
     LOGGER.debug("Setting up entry %s", entry.data)
     bridge = DynaliteBridge(hass, entry.data)
+    # need to do it before the listener
+    hass.data[DOMAIN][entry.entry_id] = bridge
     entry.add_update_listener(async_entry_changed)
```

Comment:
Why add the listener before the bridge is set up?

Reply:
For the same reason listed above. If the entry was already set up with some config, but in the next restart of HA, the configuration.yaml file has changed, the listener will need to be called as the entry is being changed and it is happening in parallel with the setup of the original entry, which has not completed yet. If I can find a better solution for the first one, will hopefully be also able to solve this

Reply:
Perhaps store the `bridge.async_setup` task, id it with the config entry id and await that before updating the entry.

Reply:
that could work. Should i store it in hass.data[entry_id]?

Reply:
Probably.

Reply:
i removed the await and it should be ok now. When I try this, it complicates things since if I await, it is still not guaranteed that the hass.data[] has been updated, depending on which gets executed first

Reply:
Let's discuss more in the PR for this.